### PR TITLE
Port changes from upstream QVTKRWI.

### PR DESCRIPTION
**Pull request by prabhuramachandran**
This ports several changes from upstream as a result this works well with PyQt6.

### CODE REVIEW

**Comment by larsoner** [comment]
The `macos-latest` failure can be fixed by using the `macos-13` image it's meant to use. `-latest` recently changed to `-14` and that's ARM64 rather than Intel so other changes are likely needed. (In principle we want both, but for expediency it's easiest to change here).

Then ideally some merge requirements could be set for the CIs (i.e., allow "merge when green" button clicking).

The latest failure is probably an issue with PySide 6.7 so you could pin to `PySide!=6.7` if you want, or try to fix it.

**Comment by prabhuramachandran** [comment]
> The `macos-latest` failure can be fixed by using the `macos-13` image it's meant to use. `-latest` recently changed to `-14` and that's ARM64 rather than Intel so other changes are likely needed. (In principle we want both, but for expediency it's easiest to change here).

Thanks, I am trying that now.
> 
> Then ideally some merge requirements could be set for the CIs (i.e., allow "merge when green" button clicking).

Yes, the interface expects some tests in the last week so I wasn't able to add those.

> 
> The other failure is probably an issue with PySide 6.7 so you could pin to `PySide!=6.7` if you want, or try to fix it.

It fails even with older versions and looks to be a bit bizarre. For now I am skipping this test with pyside6.

Can you please review the PR.

**Comment by prabhuramachandran** [comment]
@larsoner -- could you please review the PR once so I can merge this and start the release?

File: tvtk/pyface/ui/qt4/QVTKRenderWindowInteractor.py
Comment (tkoyama010) [inline tvtk/pyface/ui/qt4/QVTKRenderWindowInteractor.py:797]: Did you forget to delete debug print output?
```suggestion
```

Comment (prabhuramachandran) [reply tvtk/pyface/ui/qt4/QVTKRenderWindowInteractor.py:797]: This is part of the original upstream version and is in the `__main__` block so I think it is fine as it is convenient when debugging without having to edit the source.

Comment (tkoyama010) [reply tvtk/pyface/ui/qt4/QVTKRenderWindowInteractor.py:797]: Got it thanks!

**Review by larsoner — APPROVED** [review]
Looks good to me!

**Comment by larsoner** [comment]
I would merge, but it looks like my approval is insufficient to enable the green button

**Comment by prabhuramachandran** [comment]
> I would merge, but it looks like my approval is insufficient to enable the green button

@dpinte @mdickinson any ideas on how can we make this work?

**Comment by mdickinson** [comment]
Hi @prabhuramachandran. I've (temporarily?) disabled the "Require approvals" settings in the branch protections for master, so that this PR can be merged. We should discuss (possibly by email) a longer-term solution.

**Comment by tkoyama010** [comment]
> Hi @prabhuramachandran. I've (temporarily?) disabled the "Require approvals" settings in the branch protections for master, so that this PR can be merged. We should discuss (possibly by email) a longer-term solution.

We has set up a GitHub Bot so that when an admin user makes an LGTM comment on a PR of which they are the author, the GitHub Bot approves that PR. This allows admins to merge their PRs as they wish, while leaving the rule that admins approve other PRs.
https://github.com/pyvista/pyvista/pull/4941

**Comment by prabhuramachandran** [comment]
> Hi @prabhuramachandran. I've (temporarily?) disabled the "Require approvals" settings in the branch protections for master, so that this PR can be merged. We should discuss (possibly by email) a longer-term solution.

Thank you, yes let us discuss this via email. For now this is fine.

**Comment by prabhuramachandran** [comment]
@tkoyama010 -- Thank you for the offer. I am not sure about this and how this would work. Let me first discuss this and let you know. Thank you for the kind offer though.

**Comment by mdickinson** [comment]
> Thank you, yes let us discuss this via email.

Email sent ... Let me know if you didn't receive anything.

**Comment by prabhuramachandran** [comment]
> > Thank you, yes let us discuss this via email.
> 
> Email sent ... Let me know if you didn't receive anything.

Got it, thank you. Sorry for the slow response.

**Pull request merged**
